File: backend-spring/app-application/src/main/java/apptemplate/application/dto/notification/NotificationDto.java
package apptemplate.application.dto.notification;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class NotificationDto {

    private Long id;
    private Long userId;
    private String title;
    private String message;
    private String type;
    private String referenceId;
    private String referenceType;
    private boolean read;
    private LocalDateTime createdAt;
}

File: backend/nestjs/clean-architecture/full/src/modules/notification/application/dto/notification.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { NotificationType } from '../../domain/enums/notification-type.enum';

export class NotificationDto {
  @ApiProperty({ description: 'Notification ID' })
  id: number;

  @ApiProperty({ description: 'Notification title' })
  title: string;

  @ApiProperty({ description: 'Notification message' })
  message: string;

  @ApiProperty({ enum: NotificationType, description: 'Notification type' })
  type: NotificationType;

  @ApiProperty({ description: 'Reference entity ID', nullable: true })
  referenceId: string | null;

  @ApiProperty({ description: 'Reference entity type', nullable: true })
  referenceType: string | null;

  @ApiProperty({ description: 'Is read' })
  isRead: boolean;

  @ApiProperty({ description: 'Created at' })
  createdAt: Date;

  constructor(partial: Partial<NotificationDto>) {
    Object.assign(this, partial);
  }
}
